### PR TITLE
#113 filter expense by status

### PR DESCRIPTION
**Pull request by ddio**
## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

Issue Number: #113

## What is the new behavior?

1. Implement #113 
2. Add `/static` directory, so to create shared vue component
   1. @toomore , please help to review this. I just verify this under local dev env.
1. Now we support limited vue component feature!
    - As there's no webpack or vue-loader, plain JS file is used.
    - i.e. No `<tpl>` tag, please apply HTML in plain text.

## Other information

1. Default show `申請中` expense
   > 螢幕快照 2023-04-23 23-27-44
2. Show `目前沒有<status label>的項目 ` when no expense item under this status
   >  螢幕快照 2023-04-23 23-27-42

### CODE REVIEW

**Comment by toomore** [comment]
@ddio 看起來除了 commits 沒有 sign-off 外，其他沒什麼問題！

**Comment by ddio** [comment]
@toomore  感謝回饋！我修完之後，CICD 的機器看起來怪怪的，[memcache 裝不起來](https://github.com/COSCUP/COSCUP-Volunteer/actions/runs/4805751760/jobs/8553023224#logs)，我晚點再重跑測試看看～

**Comment by ddio** [comment]
@toomore , memcahce 裝起來後，這次卡在 `mypy` 跑不起來，應該連檢查都還沒開始。

想問你有碰過類似的問題嗎？
https://github.com/COSCUP/COSCUP-Volunteer/actions/runs/4805751760/jobs/8553545806?pr=134

**Comment by toomore** [comment]
@ddio 我終於找到問題了，flask 2.3.1 的問題。現在亮綠燈了！
https://github.com/COSCUP/COSCUP-Volunteer/pull/134/commits/84eea3511fe3bd5cbea35427ec21bcef463af231

**Comment by ddio** [comment]
@toomore  感恩！但要麻煩你 approve 呦，Github 不讓作者當 reviewer XD

File: templates/expense.html [hunk 1]
```diff
@@ -294,26 +296,55 @@ <h4>申請單狀態</h4>
 {% endblock %}
 {% block js %}
 <script src="/js/axios.min.js"></script>
+<script src="{{ url_for('static', filename='ExpenseStatusLabel.js') }}"></script>
```
Comment (toomore) [inline templates/expense.html:299]: 這行有可能在正式機有問題，但等上到正式機再來看看，我現在也想像不太出來。或是可能只能用 `/static/ExpenseStatusLabel.js`。

**Review by toomore — APPROVED** [review]
讚！

**Comment by toomore** [comment]
順便紀錄一下，bug 跑到 `pallets/werkzeug` https://github.com/pallets/werkzeug/issues/2660

**Pull request merged**